src/static/riot/analytics/analytics.tag

```diff
@@ -0,0 +1,540 @@
+            self.competitionsChart = new Chart(ctx_c, create_chart_config('# of Competitions'));
+            var ctx_s = self.refs.submission_chart.getContext('2d');
+            self.submissionsChart = new Chart(ctx_s, create_chart_config('# of Submissions'));
+            var ctx_u = self.refs.user_chart.getContext('2d');
+            self.usersChart = new Chart(ctx_u, create_chart_config('# of Users Joined'));
+
+            self.update_analytics(self.start_date, null, self.time_unit, false)
+
+        })
+
+        /*---------------------------------------------------------------------
+         Methods
+        ---------------------------------------------------------------------*/
+
+        function create_chart_config(label) {
+            return {
+                type: 'line',
+                data: {
+                    datasets: [{
+                        label: label,
+                        data: null,
+                        backgroundColor: 'rgba(54, 162, 235, 0.2)',
+                        borderColor:'rgba(54, 162, 235, 1)',
+                        borderWidth: 1,
+                        lineTension: 0,
+                    }],
+                },
+                options: {
+                    scales: {
+                        xAxes: [{
+                            type: 'time',
+                            time: {
+                                unit: 'month'
+                            }
+                        }],
+                        yAxes: [{
+                            ticks: {
+                                beginAtZero: true,
+                                stepSize: 1,
+                            }
+                        }]
+                    }
+                }
+            }
+        }
+
+        function build_chart_data(data, day_resolution, csv_format) {
+            var chart_data = []
+
+            for( let i = 0; i < data.length; i++ ) {
+                let point = {}
+                point.x = new Date(data[i]._datefield)
+                point.x.setDate(point.x.getDate() + 1)
+                point.y = data[i].count
+                chart_data.push(point)
+            }
+            chart_data.sort(function(a,b) {
+                return a.x - b.x
+            })
+            return chart_data
+         }
+
+        function update_chart(chart, data, day_resolution) {
+            chart.data.datasets[0].data = build_chart_data(data, day_resolution, false)
+            chart.update()
+        }
+
+        self.update_analytics = function (start, end, time_unit, csv) {
+
+            if (typeof start == 'string') {
+                start = self.convert_datestring_to_object(start, true)
+            }
+
+            if (typeof end == 'string') {
+                end = self.convert_datestring_to_object(end, false)
+            }
+
+            var start_date_param = new Date(start.year, start.month, start.day)
+            if (end) {
```

Comment:
can do this all in one line w/ a ternary phrase
```javascript
let end_date_param = end ? new Date(end.year, end.month, end.day) : new Date(_.now() + 86400000)
```

Reply:
Done.